src/core.js

```diff
@@ -848,7 +853,8 @@ $.extend( $.validator, {
 			// aria-describedby should directly reference the error element
 			if ( describer ) {
 				selector = selector + ", #" + describer
-					.replace( /'/g, "\\'" )
+					// See https://api.jquery.com/category/selectors/
+					.replace( /(!|"|#|\$|%|&|'|\(|\)|\*|\+|,|\.|\/|:|;|<|=|>|\?|@|\[|\\|\]|\^|`|\{|\||\}|~)/g, "\\$1" )
```

Comment:
Ah, now I get the problem. A pity we can use a closure in both methods. 

Maybe its easiesr to create a private method which returns the regex or even better one which escapes a given string?